pkg/controller/controller.go

```diff
@@ -360,42 +391,66 @@ func (c *Controller) configChanged(t *task) {
 
 	config, ok := t.obj.(*v2beta2.Config)
 	if !ok {
+		_ = level.Warn(c.logger).Log("msg", "not a config object")
 		return
 	}
 
-	tenantID := c.getTenantID(config.Labels)
-	if len(tenantID) == 0 {
+	newResourceVersion, _ := strconv.ParseUint(config.ResourceVersion, 10, 64)
+	newTenantID := c.getTenantID(config.Labels)
+	if len(newTenantID) == 0 {
 		_ = level.Warn(c.logger).Log("msg", "Ignore config because of empty tenantID", "name", config.Name, "tenantKey", c.tenantKey)
 		return
 	}
 
-	// If operation is `update` or `delete`, it needs to delete the old config.
-	if t.op == opUpdate || t.op == opDel {
-		suffix := fmt.Sprintf("/%s", config.Name)
-		for id := range c.configs {
-			found := false
-			for k := range c.configs[id] {
-				if strings.HasSuffix(k, suffix) {
+	oldTenantID := ""
+	oldResourceVersion := uint64(0)
+	suffix := fmt.Sprintf("/%s", config.Name)
+	for id := range c.configs {
+		found := false
+		for k := range c.configs[id] {
+			if strings.HasSuffix(k, suffix) {
+				oldTenantID = id
+				oldResourceVersion = c.configs[id][k].GetResourceVersion()
+				if t.op == opDel {
 					delete(c.configs[id], k)
-					found = true
 				}
+				found = true
```

Comment:
break here in the inner loop?

Reply:
If it is a delete operation, the inner loop needs to find all receivers defined in receiver crd. In the case of an update operation, the inner loop can break.